src/components/loader-spinner/loader-spinner.pw.tsx

```diff
@@ -0,0 +1,223 @@
+
+      await expect(loaderSpinnerCircleSvg(page)).toHaveCSS(
+        "stroke-width",
+        `${LOADER_SPINNER_SIZE_PARAMS[spinnerSizes].strokeWidth}px`
+      );
+    });
+  });
+
+  test("when 'showSpinnerLabel' is `true` the label is rendered", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<LoaderSpinnerComponent showSpinnerLabel />);
+
+    await expect(loaderSpinnerVisibleLabel(page)).toHaveText("Loading...");
+  });
+
+  test("when 'showSpinnerLabel' is `false` the label is not rendered", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<LoaderSpinnerComponent showSpinnerLabel={false} />);
+
+    await expect(loaderSpinnerVisibleLabel(page)).not.toBeVisible();
+  });
+
+  test("when 'showSpinnerLabel' is `false` a visually hidden alternative label is rendered", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<LoaderSpinnerComponent showSpinnerLabel={false} />);
+
+    await expect(loaderSpinnerHiddenLabel(page)).toHaveText("Loading...");
+  });
+
+  ([
+    ["action", "rgb(179, 217, 200)"],
+    ["neutral", "rgb(204, 214, 219)"],
+    ["inverse", "rgba(255, 255, 255, 0.3)"],
+    ["gradient-grey", "rgba(0, 0, 0, 0.1)"],
+    ["gradient-white", "rgb(255, 255, 255)"],
+  ] as [LoaderSpinnerColors, string][]).forEach(([colors, boxShadowValues]) => {
+    test(`when the 'color' prop is passed as '${colors}' the correct wrapper box shadow (color) value is rendered`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<LoaderSpinnerComponent color={colors} />);
+
+      const wrapperBoxShadow = await getStyle(
+        loaderSpinnerWrapper(page),
+        "box-shadow"
+      );
+
+      expect(wrapperBoxShadow).toContain(boxShadowValues);
+    });
+  });
+
+  ([
+    ["action", "rgb(0, 126, 69)"],
+    ["neutral", "rgb(51, 91, 112)"],
+    ["inverse", "rgb(255, 255, 255)"],
+    ["gradient-grey", "rgb(0, 214, 57)"],
+    ["gradient-white", "rgb(0, 214, 57)"],
+  ] as [LoaderSpinnerColors, string][]).forEach(([colors, strokeValues]) => {
+    test(`when the 'color' prop is passed as '${colors}' the correct circle svg stroke (color) value is rendered`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<LoaderSpinnerComponent color={colors} />);
+
+      await expect(loaderSpinnerCircleSvg(page)).toHaveCSS(
+        "stroke",
+        strokeValues
+      );
+    });
+  });
+
+  ([
+    ["dark", "rgba(0, 0, 0, 0.9)"],
+    ["light", "rgb(255, 255, 255)"],
+  ] as const).forEach(([labelColors, labelColorTokenValues]) => {
+    test(`when 'labelColor' is passed as '${labelColors}' the correct font color is rendered`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<LoaderSpinnerComponent labelColor={labelColors} />);
+
+      await expect(loaderSpinnerVisibleLabel(page)).toHaveCSS(
+        "color",
+        labelColorTokenValues
+      );
+    });
+  });
+});
+
+test.describe("Accessibility tests for StepFlow component", () => {
```

Comment:
```suggestion
test.describe("Accessibility tests for LoaderSpinner component", () => {
```

Reply:
Guess you guys now know which components PW file structure I copied now :0